consonance-webservice/src/main/java/io/swagger/wes/api/impl/Ga4ghApiServiceImpl.java

```diff
@@ -0,0 +1,414 @@
+        List<Job> allJobs = orderResource.listWorkflowRuns(user);
+        allJobs.stream().filter((Job t) -> String.valueOf(t.getJobId()).equals(workflowId)).forEach(
+                s -> {
+                    workflowStatus.setState(mapState(s.getState()));
+                    workflowStatus.setWorkflowId(String.valueOf(s.getJobId()));
+                });
+
+        LOG.info(workflowStatus.toString());
+
+        return Response.ok().entity(workflowStatus).build();
+    }
+
+    // If there is jobs in the queue, and running.
+    // Listing workflows will show their Id, and Status.
+    @Override
+    public Response listWorkflows(Long pageSize, String pageToken, String keyValueSearch, ConsonanceUser user) throws NotFoundException {
+        Long optionalPageSize = pageSize;
+        String optionalPageToken = pageToken;
+        String optionalKeyFilter = keyValueSearch;
+
+        Ga4ghWesWorkflowListResponse list = new Ga4ghWesWorkflowListResponse();
+
+        List<Job> allJobs = orderResource.listWorkflowRuns(user);
+        Map <Integer, JobState> serialized = new TreeMap<Integer, JobState>();
+
+        allJobs.stream().forEach((Job j) -> serialized.put(j.getJobId(), j.getState()));
+
+        for (Integer key: serialized.keySet()){
+            Ga4ghWesWorkflowDesc descriptor = new Ga4ghWesWorkflowDesc();
+            descriptor.setWorkflowId(String.valueOf(key));
+            descriptor.setState(mapState(serialized.get(key)));
+            list.addWorkflowsItem(descriptor);
+        }
+
+        return Response.ok().entity(list).build();
+    }
+
+
+     //Calls on existing method to schedule a job.
+     //Requires user with the request to be authenticated
+    @Override
+    public Response runWorkflow(Ga4ghWesWorkflowRequest body, ConsonanceUser user) throws NotFoundException, IOException, TimeoutException {
+
+        Map workflowKeyValues = body.getKeyValues();
+
+
+        String workflowTypeVersion = body.getWorkflowTypeVersion(); // Version of runner tool.
+        if (workflowTypeVersion.isEmpty()) {
+            return Response.noContent().build();
+        }
+
+        final Job newJob = new Job();
+
+        // Process deserialize body.
+        UrlValidator urlValidator = new UrlValidator();
+        String workflowDescriptor = body.getWorkflowDescriptor(); // Runner descriptor, < cwl || wdl> file.
+        if (workflowDescriptor.isEmpty()) {
+            return Response.noContent().build();
+        }
+        else if (urlValidator.isValid(workflowDescriptor)){ // A descriptor file hosted in the web.
+            URL jobURL = new URL(workflowDescriptor);
+            final Path tempFile = Files.createTempFile("image", "cwl");
+            FileUtils.copyURLToFile(jobURL, tempFile.toFile());
+            newJob.setContainerImageDescriptor(FileUtils.readFileToString(tempFile.toFile(), StandardCharsets.UTF_8));
+        }
+        else if(workflowDescriptor.indexOf("dockstore") != -1){ // A descriptor hosted in Dockstore
```

Comment:
I'd have to agree, this section seems confusing, so I'm not sure 
But checking for an index of "dockstore" seems fragile